core/src/main/java/fr/sncf/osrd/envelope_sim/IntegrationStep.java

```diff
@@ -34,10 +34,10 @@ public static IntegrationStep fromNaiveStep(
         // if the end of the step dips below 0, cut the step in half
         if (endSpeed < 0.0) {
             assert directionSign * acceleration < 0.0;
-            timeDelta = -startSpeed / (directionSign * acceleration);
+            endSpeed = 0.0;
+            timeDelta = (endSpeed - startSpeed) / (directionSign * acceleration);
```

Comment:
Why using `endSpeed` since it's value is `0.`?
```suggestion
            timeDelta = -startSpeed / (directionSign * acceleration);
```